addon/components/paper-flatpickr-input.js

```diff
@@ -114,14 +115,17 @@ export default PaperInput.extend(FlatpickrMixin, {
   },
 
   _onChange(selectedDates, dateStr, instance) {
-    //let handleInput emit onChange
+    if (this.onChange instanceof Function) {
```

Comment:
To be honest I can't remember why I let the handleInput handle the invokeAction before, haha, do you have any particular reason for doing this?

Reply:
Well.. I need different format date between the UI (the textbox) and storing them in the ember-data that I would send through the API.
I took a compare with the `ember-flatpickr` which did perfect on doing this..
So, basicly is `altFormat` is to show the formated date at the UI and `dateFormat` is to store them in the ember-data (API)

Reply:
I think I'll have to dig in again to understand this a little bit more, because I think the best way to handle the ember-data part would be through transforms, IMO, this component should be bubbling Dates or moments I always try to handle only `new Date()` or `moment()` inside ember land, and transform `(serialize/deserialize)` them inside the date transform, you can easily loose timezone data if you are handling plain date formatted strings

Reply:
Even subtle bugs if you are using moment.js 

i.e. 
```ts
let dateFormattedString = moment().format('DD/MM/YYYY')
let newDate = moment(dateFormattedString) //invalid date, moment only accepts by default MM/DD/YYYY or YYYY/MM/DD

Fix
let newDate = moment(dateFormattedString, 'DD/MM/YYYY') // moment second parameter indicates the format from the date, you would have to manually pass this around always.

I don't know if this is something way off, does it makes any sense?
```

Reply:
Yes I know that I could use the serializer to handle the data at the store for the date format; but that is just my case that need to have different `altFormat` and `dateFormat` value. Maybe there are some other cases..

Using moment would be great, but why should I reformatting if I can get the formatted date already by using the `dateFormat` params?

At `ember-flatpickr` and `flatpickr` options is support for this parameters, and I think that `ember-paper-flatpickr` could also support this too, right? Or maybe there is another way on separating this format?

Reply:
> Using moment would be great, but why should I reformatting if I can get the formatted date already by using the dateFormat params?

Because if you use this formatted string later for another date related operation, say, filtering a table by a date field, you might be surprised with bugs... my suggestion is around always use date objects while in ember, and format only when serializing.

But that being said, you are right we should just let it work as flatpickr is supposed to... I'm just not sure why changing invokeAction from handleInput to _onChange changes that?

Reply:
> > Using moment would be great, but why should I reformatting if I can get the formatted date already by using the dateFormat params?
> 
> Because if you use this formatted string later for another date related operation, say, filtering a table by a date field, you might be surprised with bugs... my suggestion is around always use date objects while in ember, and format only when serializing.

Sure, for example like filtering date range for reporting; I definitely use moment and keeping date as date data type... 

> 
> But that being said, you are right we should just let it work as flatpickr is supposed to... I'm just not sure why changing invokeAction from handleInput to _onChange changes that?

Well, I just compare this file with the file from `ember-flatpickr`, which if I don't remark the invokeAction at:

https://github.com/betocantu93/ember-paper-flatpickr/blob/8011688090cdd2b36c2fd6a59e63cfeee0105b90/addon/components/paper-flatpickr-input.js#L149-L157

, it somehow return the formatted date on the `altFormat` which maybe others need the value of `dateFormat` to consume...

Reply:
I see _why_, invokeAction inside the handleInput gets the value from the input literally i.e document.querySelector('input').value, which is ofc the altFormatted one (the pretty one), I think what I did back then was to just use the selectedDates always because of my preference/recomendation of dealing with `Date` objects inside ember... because for example in `range` mode you'd receive a string such like 

`2020-10-12 12:00 to 2020-10-21 12:00` how would you even store that in sane way? but anyway, I think your changes are okay, I also believe we need to update the addon, they are not even using the mixin I created back then